borg/platform_windows.pyx

```diff
@@ -0,0 +1,295 @@
+    ctypedef int* LPBOOL
+
+    BOOL GetFileSecurityW(LPCTSTR, int, PSID, DWORD, LPDWORD)
+    BOOL GetSecurityDescriptorOwner(PSID, PSID*, LPBOOL)
+    BOOL LookupAccountSidW(LPCTSTR, PSID, LPCTSTR, LPDWORD, LPCTSTR, LPDWORD, uint16_t*)
+    BOOL ConvertSidToStringSidW(PSID, LPCTSTR*)
+    BOOL ConvertStringSidToSidW(LPCTSTR, PSID*)
+    BOOL ConvertSecurityDescriptorToStringSecurityDescriptorW(BYTE*, DWORD, int, LPCTSTR*, int*)
+
+    cdef extern int SDDL_REVISION_1
+
+cdef extern from 'Aclapi.h':
+    ctypedef void* PACL
+    DWORD GetNamedSecurityInfoW(LPCTSTR, SE_OBJECT_TYPE, DWORD, PSID*, PSID*, PACL*, PACL*, _ACL**)
+    DWORD SetNamedSecurityInfoW(LPCTSTR, int, int, PSID, PSID, PACL, PACL)
+    DWORD SetEntriesInAclW(unsigned int, _EXPLICIT_ACCESS_W*, PACL, _ACL**)
+
+cdef PSID _get_file_security(filename, int request):
+    cdef DWORD length = 0
+    # N.B. This query may fail with ERROR_INVALID_FUNCTION
+    # for some filesystems.
+    cdef wchar_t* wcharfilename = PyUnicode_AsWideCharString(filename, NULL)
+    if wcharfilename == NULL:
+        print 'filename is NULL'
+    GetFileSecurityW(wcharfilename, request, NULL, 0, &length)
+    if GetLastError() == 122:
+        SetLastError(0)
+    else:
+        print '_get_file_security failed. Windows error:', GetLastError()
+        return NULL
+    cdef BYTE* sd = <BYTE*>malloc((length) * sizeof(BYTE))
+    GetFileSecurityW(wcharfilename, request, sd, length, &length)
+    PyMem_Free(wcharfilename)
+    return sd
+
+cdef PSID _get_security_descriptor_owner(PSID sd):
+    cdef PSID sid
+    cdef BOOL sid_defaulted
+    GetSecurityDescriptorOwner(sd, &sid, &sid_defaulted)
+    return (sid)
+
+cdef _look_up_account_sid(PSID sid):
+    cdef int SIZE = 256
+    cdef wchar_t* name = <wchar_t*>malloc((SIZE) * sizeof(wchar_t))
+    cdef wchar_t* domain = <wchar_t*>malloc((SIZE) * sizeof(wchar_t))
+    cdef DWORD cch_name = SIZE
+    cdef DWORD cch_domain = SIZE
+    cdef uint16_t sid_type = <uint16_t>0
+
+    cdef BOOL ret = LookupAccountSidW(NULL, sid, name, &cch_name, domain, &cch_domain, &sid_type)
+    if ret == 0:
+        lasterror = GetLastError()
+        if lasterror == 1332:
+            # Unknown (removed?) user or file from another windows installation
+            free(name)
+            free(domain)
+            return 'unknown', 'unknown', 0
+        else:
+            print 'Windows error:', lasterror
+
+    pystrName = PyUnicode_FromWideChar(name, -1)
+    pystrDomain = PyUnicode_FromWideChar(domain, -1)
+
+    free(name)
+    free(domain)
+    return pystrName, pystrDomain, <unsigned int>sid_type
+
+cdef sid2string(PSID sid):
+    cdef wchar_t* sidstr
+    ConvertSidToStringSidW(sid, &sidstr)
+    ret = PyUnicode_FromWideChar(sidstr, -1)
+    LocalFree(<HLOCAL>sidstr)
+    return ret
+
+def get_owner(path):
+    cdef int request = OWNER_SECURITY_INFORMATION
+    cdef BYTE* sd = _get_file_security(path, request)
+    if sd == NULL:
+        return 'unknown', 'unknown'
```

Comment:
The first is human name, the secont the SID? Then the second should be the null SID instead of "unknown" I think.